environs/manual/winrmprovisioner/provisioner.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016 Canonical Ltd.
+// Copyright 2016 Cloudbase Solutions SRL
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package winrmprovisioner
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/environs/manual"
+)
+
+var logger = loggo.GetLogger("juju.environs.manual.winrmprovisioner")
+
+// ProvisionMachine returns a new machineId and nil if the provision process is done successfully
+// The function will manual provision a windows machine using as comunication protocol WinRM(windows remote manager)
+func ProvisionMachine(args manual.ProvisionMachineArgs) (machineId string, err error) {
+	defer func() {
+		if machineId != "" && err != nil {
+			logger.Errorf("provisioning failed, removing machine %v: %v", machineId, err)
+			if cleanupErr := args.Client.ForceDestroyMachines(machineId); cleanupErr != nil {
+				logger.Errorf("error cleaning up machine: %s", cleanupErr)
+			}
+			machineId = ""
+		}
+	}()
+
+	if err = InitAdministratorUser(&args); err != nil {
+		return machineId, errors.Annotatef(err,
```

Comment:
Have you pushed your fixes? I'm pretty sure Nate meant to use a string literal of `""` here (i.e. `return "", ...`).